### PR TITLE
Fix long long json serialization

### PR DESCRIPTION
**Pull request by crazyhappygame**
Problem:
long long was serialized as string in json. 

```
cereal/unittests/pod.hpp:51: ERROR: CHECK_EQ( os1.str(), os2.str() ) is NOT correct!
   values: CHECK_EQ( {
     "value0": -96
 }, {
     "value0": "-96"
 } )
```

Comments:
It worked fine for MSVC. long long and int64_t are the same types
```
typedef long long          int64_t;
```
It does not work for GCC/clnag

### CODE REVIEW

**Comment by crazyhappygame** [comment]
There are many issues reported for this problem:
https://github.com/USCiLab/cereal/issues/540
https://github.com/USCiLab/cereal/issues/556

@AzothAmmo could you merge it?

**Comment by crazyhappygame** [comment]
Hello @AzothAmmo, could this simple fix be added to the v1.3.1 release?

File: unittests/pod.hpp [hunk 1]
```diff
@@ -144,4 +144,73 @@ void test_pod()
...
+    oar(o_t1);
+  }
+
+  std::ostringstream os2;
+  {
+    OArchive oar(os2);
+    oar(o_t2);
+  }
+
+  CHECK_EQ(os1.str(), os2.str());
```
Comment (crazyhappygame) [inline unittests/pod.hpp:166]: Before fix this check failed
```
cereal/unittests/pod.hpp:166: ERROR: CHECK_EQ( os1.str(), os2.str() ) is NOT correct!
   values: CHECK_EQ( {
     "value0": -96
 }, {
     "value0": "-96"
 } )
```

**Comment by crazyhappygame** [comment]
@AzothAmmo I got message "First-time contributors need a maintainer to approve running workflows". Could you approve?

**Comment by AzothAmmo** [comment]
> @AzothAmmo I got message "First-time contributors need a maintainer to approve running workflows". Could you approve?

You should be able to run this against CI now.

**Comment by crazyhappygame** [comment]
@AzothAmmo could you move this PR forward and merge it?

File: unittests/pod.hpp [hunk 2]
```diff
@@ -144,4 +144,73 @@ void test_pod()
   }
 }
 
+template <class IArchive, class OArchive, class T1, class T2>
+void test_pod_serialization(std::mt19937& gen)
+{
+  T1 const o_t1 = random_value<T1>(gen);
+  T2 const o_t2 = o_t1;
+  INFO(std::string("typeid(int).name(): ") + typeid(T2).name());
```
Comment (AzothAmmo) [inline unittests/pod.hpp:152]: this seems to be causing all of the MSVC test cases to fail

**Pull request merged**